pkg/resources/instance_pool/datasource.go

```diff
@@ -205,58 +216,68 @@ func dsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.
 		)
 	}
 
-	pool, err := client.FindInstancePool(
-		ctx,
-		zone, func() string {
-			if byID {
-				return id.(string)
+	var pool *v3.InstancePool
+	if byID {
+		pool, err = client.GetInstancePool(
+			ctx,
+			v3.UUID(id.(string)),
+		)
+	} else {
+		rep, err := client.ListInstancePools(ctx)
+		if err == nil {
+			return diag.FromErr(err)
+		}
+		for _, p := range rep.InstancePools {
+			if p.Name == name {
+				pool = &p
+				break
```

Comment:
There is also FindInstancePool helper: https://github.com/exoscale/egoscale/blob/master/v3/operations.go#L10016

Reply:
Fixed in [c3c4b9d](https://github.com/exoscale/terraform-provider-exoscale/pull/406/commits/c3c4b9dc5f133e03abe934db45ef4ab939dede1c)